src/Microsoft.AspNetCore.Server.Kestrel.Configuration/KestrelConfigurationExtensions.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Options.Infrastructure;
+
+namespace Microsoft.AspNetCore.Hosting
+{
+    public static class KestrelConfigurationExtensions
+    {
+        /// <summary>
+        /// Specify Kestrel as the server to be used by the web host and bind its settings from configuration.
+        /// </summary>
+        /// <param name="hostBuilder">
+        /// The Microsoft.AspNetCore.Hosting.IWebHostBuilder to configure.
+        /// </param>
+        /// <returns>
+        /// The Microsoft.AspNetCore.Hosting.IWebHostBuilder.
+        /// </returns>
+        public static IWebHostBuilder UseKestrelWithConfiguration(this IWebHostBuilder hostBuilder)
```

Comment:
Is there no way to takeover the default UseKestrel() which would make it match all of the other AddXyz()'s that also add default AddConfigureOptions?

Can we just add a reference to Microsoft.AspNetCore.Certificates.Configuration.Sources and ConfigurationBinder to Microsoft.AspNetCore.Server.Kestrel and do this inside of UseKestrel?